src/buf/vec_deque.rs

```diff
@@ -1,3 +1,6 @@
+#[cfg(not(feature = "std"))]
+use alloc::collections::VecDeque;
+#[cfg(feature = "std")]
 use std::collections::VecDeque;
```

Comment:
`std` is just re-exporting the `VecDeque` type from `alloc`, so this sort of gating isn't really necessary

Reply:
@tarcieri i agree with you, but the goal was to address

> I believe we don't need to link against alloc when std is already present

if in `std` mode we don't want to refer to the `alloc` crate at all, then this is the pattern we have to use, AFAIU

Reply:
the lines you are commenting were added in @DoumanAsh commit: https://github.com/tokio-rs/bytes/pull/281/commits/2f3a5b0982bb0a29abe61942585d4bda6c5cb0a4

Reply:
lmk if you both think it's better to just use `no_std` and `extern crate alloc;` in all configurations, with `extern crate std` as well in std configuration, i can change it back to that way

Reply:
What's the rationale for that?

Reply:
i suspect that there is ultimately no difference either way

Reply:
@cbeck88 that'd be my preference. Unless you're trying to support MSRV < 1.36 I don't see the point in doing otherwise

Reply:
@DoumanAsh do you agree?

Reply:
@DoumanAsh i think it's also worth noting that the RFC that stabilized alloc crate seems to say that we should prefer that pattern:

https://rust-lang.github.io/rfcs/2480-liballoc.html

i.e. they say to do 

```
#![no_std]

extern crate alloc;

use alloc::vec::Vec;
```

instead of

```
#![no_std]

#[cfg(feature = "no_std")] extern crate alloc;
#[cfg(not(feature = "no_std"))] extern crate std as alloc;

use alloc::vec::Vec;
```

Reply:
Ok, if RFC was stabilized in 1.36 already then we're good to use it